src/rust/engine/Cargo.toml

```diff
@@ -66,10 +68,12 @@ boxfuture = { path = "boxfuture" }
 enum_primitive = "0.1.1"
 fnv = "1.0.5"
 fs = { path = "fs" }
-futures = "0.1.16"
+futures = "0.1"
```

Comment:
We actually have a hard dep on 0.1.16 as a minimum; I remember bumping it at some point. I just tried forcing 0.1.15 and the build failed, whereas forcing 0.1.16 was successful.

I've been meaning to write a little utility or something which tries to build a crate actually forcing all deps to their minimum specified compatible version...

Reply:
I needed to remove this constraint in order to not see a failure where tokio selected one version (`0.1.21`) and our explicitly selected version interfered. But I think it should work with a "caret requirement"? https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html